connect/runtime/src/main/java/org/apache/kafka/connect/runtime/standalone/StandaloneHerder.java

```diff
@@ -427,9 +427,8 @@ private void updateConnectorTasks(String connName) {
         }
 
         List<Map<String, String>> newTaskConfigs = recomputeTaskConfigs(connName);
-        List<Map<String, String>> oldTaskConfigs = configState.allTaskConfigs(connName);
 
-        if (!newTaskConfigs.equals(oldTaskConfigs)) {
```

Comment:
This logic seems significantly simpler than what's currently used in `DistributedHerder:: reconfigureConnector ` and moved to `AbstractHerder::taskConfigsChanged` in this PR.

Why not keep this logic instead of removing `ConfigClusterState::allTaskConfigs`?

Reply:
This logic is simpler, but only because `ClusterConfigState::allTaskConfigs` and `List::equals` are abstracting some of the complexity.

There are multiple reasons I decided to keep the DistributedHerder implementation instead of the Standalone herder:

1. The DistributedHerder implementation allows us to provide more detailed debug logs about how the task configs are different, in a way that `List::equals` hides. Since distributed mode is more common, I figured adding those logs to standalone was better than taking them away from distributed mode.
2. The `taskConfig` and `allTaskConfigs` are duplicating one another to some degree (they're both retrieving task configs, they're both applying transformations) but we cannot remove `taskConfig` as it is used in lots of other areas of the herder.
3. The `allTaskConfigs` is currently performing a naive search over all known task configs, and while we could fix that, it requires a larger change with higher regression risk.

Reply:
I think the most convincing argument here is point 1; any duplication of work that isn't inevitable can be addressed to the same degree regardless of which class contains the logic, and I don't think the risk of improving the naive search is super high.

Anyways, sounds good to me 👍